File: web-vue/src/types/dataset.ts
/**
 * 数据集
 */
export interface Dataset {
    /**
     * 数据集ID
     */
    id: number
    
    /**
     * 数据集名称
     */
    name: string
    
    /**
     * 用户ID
     */
    userId: number
    
    /**
     * 存储桶
     */
    bucket: string
    
    /**
     * 对象键
     */
    objectKey: string
    
    /**
     * 数据集描述
     */
    description: string
    
    /**
     * 状态(0:禁用 1:启用)
     */
    status: number
    
    /**
     * 创建时间
     */
    createTime: string
    
    /**
     * 更新时间
     */
    updateTime: string
}

/**
 * 数据集分页列表
 */
export interface DatasetPage {
    /**
     * 记录列表
     */
    records: Dataset[]
    
    /**
     * 总记录数
     */
    total: number
    
    /**
     * 每页记录数
     */
    size: number
    
    /**
     * 当前页码
     */
    current: number
} 

File: web-springboot/src/main/java/com/example/entity/Dataset.java
package com.example.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 数据集实体类
 */
@Data
@TableName("training_dataset")
public class Dataset {
    
    /**
     * 数据集ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;
    
    /**
     * 用户ID
     */
    private Long userId;
    
    /**
     * 数据集名称
     */
    @TableField("dataset_name")
    private String name;
    
    /**
     * 数据集存储路径
     */
    @TableField("dataset_path")
    private String filePath;
    
    /**
     * 数据集大小(字节)
     */
    @TableField("dataset_size")
    private Long fileSize;
    
    /**
     * 数据集描述
     */
    private String description;
    
    /**
     * 状态：0-无效，1-有效
     */
    private Integer status;
    
    /**
     * 创建时间
     */
    private LocalDateTime createTime;
    
    /**
     * 更新时间
     */
    private LocalDateTime updateTime;
    
    /**
     * 是否删除：0-未删除，1-已删除
     */
    @TableLogic
    private Integer deleted;
}